Doms gifts

DominiekMeers opened the pull request:
Added completely functional wishlist
Added support fot userId
Stripped some unneeded code

DominiekMeers committed:
Various updates

Wishlist functionality is done. Need to fix flow issues and look of the pages

DominiekMeers committed:
added UserId/fixed flow

- Added support for UserId session variable
- fixed wishlist flow
- converted wishlist files from js to tsx

DominiekMeers committed:
Working wishlist!

Cleanup: deleted temp and debug files.
Added support for userId in the app.
Added support for giftId in states needed in the app.
Fixed flow issues and cleaned up modules

ailpon reviewed (commented):
awesome job Dom

gdijkhoffz commented on giftree/src/pages/AddGift.tsx:
```diff
@@ -0,0 +1,171 @@
+    setGiftAdded(true);
+  };
+
+  if (giftAdded)
```
Regarding the bug that doesn't take you to the "add gift" page after pressing the "Add another gift button" might be because you are not resetting the `giftAdded` variable so whenever it re-renders this page, `giftAdded` is still true, so then it goes into this if statement again.

gdijkhoffz commented:
Now that I've gone into the branch and tested the new features I found a few issues:

- If you add a gift, when presented with the screen with the buttons "add another gift" and "wishlist", when I press the "add another gift" button it goes back to the "add gift screen" for a split second, and then goes back to the screen showing the "add another gift" and "wishlist" buttons. There seems to have a bug here when trying to go back to the add gift screen, please try to investigate because it's probably a quick fix. Another option would also be to just add the gift and automatically go back to the wishlist and then the user can just click add gift again, this would also be a smoother process in general.
- The "add gift" page allows the user to press "add gift" button without filling in the form. Please take a look at my code for login and signup and you can add the prevention check for submitting an empty form, as well as the alerts to the user letting them know the fields are required. I'm thinking everything should be required except for the comment.
- The delete gift is a little weird imo. I think it would be better to have it do a "confirm delete" pop up or modal, then when you click ok it automatically goes back to the wishlist, instead of having to press the "wishlist" button to go back. If the user click can just remove the pop up and continue to show the edit mode for the gift.

Other than these minor issues, everything seems to be working well. Great job completing most of this feature, if you can just address the issues above, then I can approve this.

DominiekMeers committed:
Fixed bugs + improved flow

Fixed a bug in the add gift page.
Added error checking for empty strings.
Improved flow for edit gifts.
Added a popup warning before deleting gifts.

DominiekMeers committed:
cleanup - group functionality - bug fixes

Removed unnecessary code
Added functionality for groups
fixed bugs in wishlist
improved flow: when adding/deleting gifts and groups you get redirected back to the lists

gdijkhoffz committed:
merge and fix conflicts with main

gdijkhoffz committed:
Change endpoints and add redirect after giftupdate

DominiekMeers committed:
Merge branch 'main' into DomsGifts

DominiekMeers committed:
Minor cleanup
